packages/web/src/components/molecules/ExecutionStepsList/ExecutionStepsList.tsx

```diff
@@ -67,18 +64,15 @@ const ExecutionStepsList: FC<ExecutionStepsListProps> = props => {
             };
           })
           .map(({status, icon, name, url}, index) => (
-            <ExecutionStepsListItemExecution
-              // eslint-disable-next-line react/no-array-index-key
-              key={`item-${index}`}
-              className={classNames({clickable: url})}
-              onClick={url ? () => navigate(url) : undefined}
-            >
-              <StyledSpace size={15}>
-                {status ? <StatusIcon status={status} /> : null}
-                {icon}
-                <ExecutionName name={name} />
-                {url ? <StyledExternalLinkIcon /> : <div />}
-              </StyledSpace>
+            <ExecutionStepsListItemExecution key={url ?? index} className={classNames({clickable: url})}>
```

Comment:
* `url` is not necessarily unique - you may have the same test run multiple times in a single step, so, unfortunately, we have to base `key` on `index`
* Nested `NavLink` changes the design a little - could we i.e. change `ExecutionStepsListItemExecution` to be styled `NavLink` instead?

```tsx
const ExecutionStepsListItemExecution = styled(NavLink)`
  the styles goes here
`;
```